### PR TITLE
Fix issue installing libsystemd-dev on Debian 8.

### PR DESCRIPTION
**Pull request by mikemadden42**
The libsystemd0 package is also required.

### CODE REVIEW

File: journalbeat/magefile.go [hunk 1]
```diff
@@ -188,7 +189,7 @@ func installLinuxS390X() error {
 	return installDependencies(libsystemdDevPkgName+":s390x", "s390x")
 }
 
-func installDependencies(pkg, arch string) error {
+func installDependencies(arch string, pkgs ...string) error {
```
Comment (urso) [inline journalbeat/magefile.go]: this PR changes order of parameters, each call to 'installDependencies' must be modified.

File: journalbeat/magefile.go [hunk 2]
```diff
@@ -200,7 +201,11 @@ func installDependencies(pkg, arch string) error {
 		return err
 	}
 
-	return sh.Run("apt-get", "install", "-y", "--no-install-recommends", pkg)
+	if arch == "i386" {
+		return sh.Run("echo", "'Yes, do as I say!'", "|", "apt-get", "install", "-y", "--force-yes", "--no-install-recommends", pkgs...)
+	} else {
```
[the remark this answers is not in the document]
Comment (mikemadden42) [reply journalbeat/magefile.go]: I dropped the `else`.

File: journalbeat/magefile.go (hunk 1 above)
Comment (mikemadden42) [reply journalbeat/magefile.go]: I swapped the arch & pkg.

File: journalbeat/magefile.go [hunk 3]
```diff
@@ -200,7 +201,12 @@ func installDependencies(pkg, arch string) error {
 		return err
 	}
 
-	return sh.Run("apt-get", "install", "-y", "--no-install-recommends", pkg)
+	if arch == "i386" {
```
Comment (andrewkroh) [inline journalbeat/magefile.go]: Why the special case of running a script for i386? Seems like you should be able to run a different command (or with different params) directly from Go.

File: journalbeat/install_pkgs.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh -x
+
+echo 'Yes, do as I say!' | echo apt-get install -y --force-yes --no-install-recommends $*
```
Comment (andrewkroh) [inline journalbeat/install_pkgs.sh]: `echo` piped to `echo`? Is this doing something I don't understand? :)

File: journalbeat/magefile.go (hunk 3 above)
Comment (urso) [reply journalbeat/magefile.go]: We need to send `Yes, do as I say!` to stdin of apt-get. Otherwise installation fails. Even if `--force-yes` is used.

Comment (urso) [reply journalbeat/magefile.go]: With recent changes to debian 8 packages one can not install `libsystemd-dev:i386` without `libsystem0:i386` anymore :(

It's a workaround for now.

Yet I wonder if that's the case for i386 only or other architectures as well.

File: journalbeat/install_pkgs.sh (hunk 4 above)
Comment (urso) [reply journalbeat/install_pkgs.sh]: indeed, should be `echo '...' | apt-get`

**Comment by urso** [comment]
@andrewkroh might be related to this: https://www.linuxcompatible.org/news/printer/systemd_and_putty_updates_for_debian_8_lts.html

File: journalbeat/magefile.go (hunk 3 above)
Comment (andrewkroh) [reply journalbeat/magefile.go]: I'd prefer to not have the external script, but no need to address this with it being a quick fix and hopefully temporary once https://github.com/elastic/golang-crossbuild/issues/23 is addressed.

The script could probably be avoided by either running something like `sh.Run("/bin/sh", "-c", ...)` or by directly using `exec.Command()` and writing the necessary string to stdin.

**Comment by mikemadden42** [comment]
Quoting @urso 

> Looks like it’s fixed by now by debian package maintainers. When checking our workaround we noticed that the version has been changed from `215-17+deb8u12` to `215-17+deb8u13` and our workaround was not needed anymore.

**Comment by andrewkroh** [comment]
Closing because this workaround is no longer necessary. The upstream debian package was fixed.

**Pull request closed without merging**